runtime/gc_glue_java/CollectorLanguageInterfaceImpl.cpp

```diff
@@ -700,13 +700,20 @@ MM_CollectorLanguageInterfaceImpl::scavenger_switchConcurrentForThread(MM_Enviro
 			j9tty_printf(PORTLIB, "%p: Nursery [%p,%p] Evacuate [%p,%p] GS [%p,%p] Section size 0x%zx, sections %lu bit offset %lu bit mask 0x%zx\n",
 					vmThread, nurseryBase, nurseryTop, gsBase, gsTop, pageBase, pageTop, _extensions->getConcurrentScavengerPageSectionSize() , sectionCount, startOffsetInBits, bitMask);
 		}
+#if defined(J9VM_GC_COMPRESSED_POINTERS) && defined(J9VM_ARCH_X86)
+		vmThread->evacuateBase = (void*)(((uintptr_t)gsBase) >> vmThread->javaVM->compressedPointersShift);
+		vmThread->evacuateTop = (void*)(((uintptr_t)gsBase) >> vmThread->javaVM->compressedPointersShift);
```

Comment:
Why don't the calls to get the initial value just return this constant?  Then the z an x code can be the same:

ie: `void* gsBase = _extensions->scavenger->getEvacuateBase();`

Reply:
Updated to keep both uncompressed and compressed ranges, so that X and Z code are now the same.